src/data-extractor/javaDataExtractor.ts

```diff
@@ -47,8 +47,10 @@ export class JavaDataExtractor extends AbstractDataExtractor {
     return { type: variable.type, name: variable.name, value: variable.value };
   }
 
-  createVariableId(variable: DebugProtocol.Variable): string {
-    return this.isNullVariable(variable) ? addNullPrefix(hash(variable)) : addObjectPrefix(variable.value.split(this.sizeSuffix)[0]);
+  createVariableId(variable: DebugProtocol.Variable, parentId?: string | undefined): string {
+    return this.isNullVariable(variable)
+      ? addNullPrefix(hash(parentId ? parentId : variable))
```

Comment:
If an object has 2 references to null, the null nodes would have the same id. Maybe you could add the name of the reference to the id. E.g., ``` `${parentId}_${variableName}` ```